File: src/main/java/com/google/service/UserService.java
package com.google.service;

import com.google.domain.User;
import com.googlecode.objectify.Key;
import com.googlecode.objectify.NotFoundException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

import static com.googlecode.objectify.ObjectifyService.ofy;

@Service
public class UserService {

    private static final Logger logger = LoggerFactory.getLogger(UserService.class);

    public User saveUser(User user){
        logger.info("Save user");
        try {
            ofy().save().entity(user).now();
            return user;
        } catch (Exception e){
            throw e;
        }
    }

    public User getUser(String id){
        logger.info("Get user");
        User user;
        try {
            user = ofy().cache(false).load().type(User.class).id(id).now();
            if (user == null){
                throw new NotFoundException();
            }
            else {
                return user;
            }
        } catch (Exception e){
            throw e;
        }
    }

}

File: src/main/java/com/google/service/UserService.kt
package com.google.service

import com.google.domain.User
import com.googlecode.objectify.Key
import com.googlecode.objectify.NotFoundException
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

import com.googlecode.objectify.ObjectifyService.ofy

@Service
class UserService {

    private val logger = LoggerFactory.getLogger(UserService::class.java)

    fun saveUser(user: User): User {
        logger.info("Save user")
        try {
            ofy().save().entity(user).now()
            return user
        } catch (e: Exception) {
            throw e
        }

    }

    fun getUser(id: String): User {
        logger.info("Get user")
        val user: User?
        try {
            user = ofy().cache(false).load().type(User::class.java).id(id).now()
            return user ?: throw NotFoundException()
        } catch (e: Exception) {
            throw e
        }

    }

    fun getUsersById(userIdsList: List<String>): List<User>{
        logger.info("Get user list")
        val userList: List<User>?
        try {
            userList = ofy().load().type(User::class.java).ids(userIdsList).values.toList()
            return userList ?: throw NotFoundException()
        } catch (e: Exception) {
            throw e
        }
    }



}
